### DISCUSSION TITLE
JED - Download Links

### DISCUSSION THREAD

**Discussion by PhocaCz** [joomla/joomla-cms#35641] [General]
As I do not have enough information, the following questions may be based more on my ignorance, so I apologize in advance if I am asking about something that has already been resolved or is being resolved.

Once again, I am managing two active versions of an extension at the same time. One for J3 and one for J4. However, on the JED I only see a field for the download link (one file) and I don't see a field for entering the manifest.xml. The problem is not with manual installation, one visits the download link and selects the correct version there. However, the problem is when the list of extensions is displayed directly in the CMS. There the wrong or missing link is already a problem.

I guess there is some work on a new version of JED. This feature is probably planned. However, J4 already exists and now the current situation is that it is impossible to prepare the right version of the extension for the right version of the CMS.

Still consider that I may be wrong and just lack knowledge of JED, so the question is:

a) Is there such an option to insert a manifest.xml file on JED, if so, where?
b) If not, what are the plans?

Thank you, Jan

**Comment by wojsmol** [comment]
> a) Is there such an option to insert a manifest.xml file on JED, if so, where?

Do you referring to updateserver definition - something like [this](https://github.com/backbutton/packages/blob/master/manifest.xml)?

**Comment by PhocaCz** [comment]
Yes, I mean standard manifest.xml which will provide right extension version for right CSM version.

**Comment by Bakual** [comment]
You can put the update manifest into the "Download URL" field. This should work fine for multiple versions.
![image](https://user-images.githubusercontent.com/1018684/134357034-cb46a306-3e82-4a78-8959-476ad65ddc75.png)

**Comment by PhocaCz** [comment]
Is there some cache feature, etc?

When adding manifest.xml

![img](https://i.imgur.com/99FeAky.png)

nothing changes in the list in CMS:
![img](https://i.imgur.com/dqQ8N7z.png)

![img](https://i.imgur.com/22toik4.png)

**Reply by Bakual** [reply]
The first versions of the Install From Web used an own server to deliver the content of the Webinstaller. There sure was a cache involved as the data wasn't directly from JED.
I don't know if that is still the case. But I certainly would expect a cache.
